Tasks/DotNetCoreInstallerV1/Task Design.md

```diff
@@ -0,0 +1,110 @@
+    - NOTE: in case exact version is not found, the function logs error and task will fail.
+  - x : in this case, we find the latest channel with `support-phase` != `preview` (unless `includePreviewVersions` is true). The releases.json for the selected channel is then downloaded and parsed into json object `releaseInformation`.
+    - The version with heighest `release.sdk/runtime.version` is then returned. (preview version will only be selected if `includePreviewVersions` is true)
+  - Empty: the function will log error as version needs to be given till atleast minor version. As a result the task will fail.
+
+
+- Based on Patch version value, the following can happen
+  - Empty : the function logs error and task will fail as it expects either a value or x.
+  - Exact Version (alpha numeric): a release with exact release.sdk/runtime.version will be searched in `channelInformation`.
+    - `version found:` the version information is returned
+    - `version not found:` function logs error and task will fail.
+  - x : The latest sdk/runtime version among all releases in that channel will be searched. The latest version information is then returned.
+
+## **- Support for multiple versions to be installted side by side**
+This feature was asked by users so that multiple versions can be installed using the task and they all will be available to subsequent tasks in pipeline. It is required in projects where multiple versions of SDK/runtime are required to build/run different applications.
+
+Advantages of this feature:
+- User can build all applications in a project using a single build task and need not create different pipelines or add multiple build task.
+
+### How it works:
+This feature is implemented, by asking user a new input: `installationPath`, the path where .Net core SDK/runtime should be installed. The Agent process should have write access on the path.
+There are multiple benefits of asking `installationPath` from user:
+- User can choose to install a particular version in isolation of all other version already installed, by specifying a path which is cleaned up after each release/build.
+- User can augment existing set of versions installed with another version by specifying a path where a set of .Net core versions might already be installed, such as: $(Agent.ToolsDirectory)
+- User can also install it at global .Net Core installation path, such as: C:\Program Files\dotnet or /usr/share/dotnet/. This way all agents running on that machine will have access to the newly added .Net core versions.
+
+The way a version is installed at the `installationPath` is now as follows:
+- We extract all folders from dotnetVersions.zip/tgz into the `installationPath`, just like install-dotnet.ps1/sh.
+- The files in root directory of archive are only copied/overriden if
+  - The .Net core version being installed at `installationPath` is later than all other versions already installed at the path.
```

Comment:
I assume failure in copying these top level files will not cause task failure

Reply:
Nope !